core/code/ornaments.js

```diff
@@ -13,52 +13,120 @@
 // - Beacons (pe$TAG - $NAME) ie: 'peNIA - NIANTIC'
 // - Frackers ('peFRACK')
 // (there are 7 different colors for each of them)
+//
+// Ornament IDs are dynamic. NIANTIC might change them at any time without prior notice.
+// New ornamnent IDs found on the map will be recorded and saved to knownOrnaments from
+// which the Ornaments dialog will be filled with checked checkboxes.
+// To exclude a set of ornaments, even if they have not yet shown up on the map, the user
+// can add an entry to excludedOrnaments, which will compared (startsWith) to all known and
+// future IDs. example: "ap" to exclude all Ornaments for anomalies (ap1, ap2, ap2_v)
 
+/* global L, dialog */
 
 window.ornaments = {
 
   OVERLAY_SIZE: 60,
   OVERLAY_OPACITY: 0.6,
+  iconUrls: [],
+  icon:[],
+  excludedOrnaments: [],
+  knownOrnaments: {},
 
   setup: function () {
     this._portals = {};
-    var layerGroup = L.layerGroup;
+    window.ornaments.layerGroup = L.layerGroup;
     if (window.map.options.preferCanvas && L.Browser.canvas && !window.DISABLE_CANVASICONLAYER) {
-      layerGroup = L.canvasIconLayer;
+      window.ornaments.layerGroup = L.canvasIconLayer;
       L.CanvasIconLayer.mergeOptions({ padding: L.Canvas.prototype.options.padding });
     }
-    this._layer = layerGroup();
-    this._beacons = layerGroup();
-    this._frackers = layerGroup();
+    this.load();
+
+    this._layer = window.ornaments.layerGroup();
+    this._excluded = window.ornaments.layerGroup(); // to keep excluded ornaments in an own layer
+
     window.layerChooser.addOverlay(this._layer, 'Ornaments');
-    window.layerChooser.addOverlay(this._beacons, 'Beacons');
-    window.layerChooser.addOverlay(this._frackers, 'Frackers');
+    window.layerChooser.addOverlay(this._excluded, 'Excluded', {default: false});
+
+    $('<a>')
+      .html('Ornaments Opt')
+      .attr({
+        id: 'ornaments-toolbox-link',
+        title: 'Edit ornament exclusions',
+        accesskey: 'o'
+      })
+      .click(window.ornaments.ornamentsOpt)
+      .appendTo('#toolbox');
```

Comment:
I personally prefer the compact defintion
```suggestion
    $('<a>', {
     text:'Ornaments Opt',
     id: 'ornaments-toolbox-link',  
     title: 'Edit ornament exclusions',
     accesskey: 'o',
     click: window.ornaments.ornamentsOpt})
      .appendTo('#toolbox');
```

BTW: is "id" used?

Reply:
which one is used in IITC?

Reply:
IITC has no rule here.  Every dev had his own way to do it.

as said both ways are fine. I only switched "html" to "text" here.

Reply:
accepted to new PR